### DISCUSSION TITLE
Is it normal when passing a baseUrl to "createWebHistory" to have a trailing slash?

### DISCUSSION THREAD

**Discussion by BrachyKnight** [vuejs/router#1835] [Help and Questions] [answered]
Hi,
I am writing this question since I noticed that when I create a router by passing it `createWebHistory(baseUrl)` it always appends a trailing slash. So if I pass the url "settings", it always redirects me to "settings/". This happens only on the baseUrl and not on its children.
Is this behavior intended? Is there a way to change/prevent this?
I am using vue router 4.1.6

**Comment by BrachyKnight** [comment]
I have an isasue with this behaviour since I am working on a MFE project that uses React as well, and its router doesn't append a / to the base Url.
Another issue that I've noticed is how the vue-router treats the baseUrl in the history. If I pass a baseUrl of "settings" and I navigate to "settings/", the history.state object doesn't have absolute paths: current will be "/" and if I navigate to "settings/privacy" current will be just "/privacy". This messes up history when working in a MFE context, because if I navigate to another domain (this doesn't trigger a page reload since I am using single-spa) and go back using the browser arrow it will navigate to "/"

**Comment by posva** [comment, marked as answer]
I think you might not need `base` in your project, [it has a different purpose](https://router.vuejs.org/api/interfaces/RouterHistory.html#Properties-base)

**Reply by BrachyKnight** [reply]
I need something like `base` since I need a way to tell vue apps in which domain they have been mounted. Is there a way to tell the router to not append the / on the home route?

**Reply by posva** [reply]
That’s not possible, it’s by design to keep routing consistent with and without a base

**Reply by BrachyKnight** [reply]
Thank you, I understand now why it has been developed like that. Is there some documentation on how `vue-router` manages and reads history, in particular the `state` property? I have tried to intercept the url changes so that I can use `replaceHistory` to make `vue-router` work like i would like it to, but I don't know which properties to touch or if the route uses some internal variables which I have no access to
